WooCommerce/Classes/POS/Presentation/Reusable Views/Buttons/POSButtonStyle.swift

```diff
@@ -0,0 +1,175 @@
+import SwiftUI
+
+/// The visual variant of the POS button.
+enum POSButtonVariant {
+    case filled
+    case outlined
+}
+
+/// The size variant of the POS button.
+enum POSButtonSize {
+    case normal
+    case extraSmall
+}
+
+/// A unified button style for POS that supports different variants and sizes.
+struct POSButtonStyle: ButtonStyle {
+    @Environment(\.colorScheme) var colorScheme
+    @Environment(\.isEnabled) var isEnabled
+
+    let variant: POSButtonVariant
+    let size: POSButtonSize
+
+    init(variant: POSButtonVariant = .filled,
+         size: POSButtonSize = .normal) {
+        self.variant = variant
+        self.size = size
+    }
+
+    func makeBody(configuration: Configuration) -> some View {
+        Group {
+            switch size {
+            case .normal:
+                HStack {
+                    Spacer()
+                    configuration.label
+                    Spacer()
+                }
+            case .extraSmall:
+                configuration.label
+            }
+        }
+        .padding(.vertical, size.padding.vertical)
+        .padding(.horizontal, size.padding.horizontal)
+        .font(size.font)
+        .background(backgroundColor)
+        .foregroundColor(foregroundColor)
+        .overlay(borderOverlay)
+        // Makes the entire area tappable, otherwise the area with clear background is not tappable.
+        .contentShape(Rectangle())
+        .cornerRadius(Constants.cornerRadius)
+        .opacity(configuration.isPressed ? 0.7 : 1.0)
+        .animation(.easeOut(duration: 0.15), value: configuration.isPressed)
+    }
+
+    private var backgroundColor: Color {
+        switch (variant, isEnabled) {
+        case (.filled, true):
+                .posPrimaryContainer
+        case (.filled, false):
+                .posDisabledContainer
+        case (.outlined, _):
+                .clear
+        }
+    }
+
+    private var foregroundColor: Color {
+        switch (variant, isEnabled) {
+        case (.filled, true):
+                .posOnPrimaryContainer
+        case (.filled, false):
+                .posOnDisabledContainer
+        case (.outlined, true):
+                .posOnSurface
+        case (.outlined, false):
+                .posOnDisabledContainer
+        }
+    }
+
+    @ViewBuilder
+    private var borderOverlay: some View {
+        if variant == .outlined {
+            RoundedRectangle(cornerRadius: Constants.cornerRadius)
+                .stroke(isEnabled ? Color.posInverseSurface : .posDisabledContainer,
+                        lineWidth: Constants.borderStrokeWidth)
+        }
+    }
+}
+
+// MARK: - POSButtonStyle Constants
+
+private extension POSButtonStyle {
+    enum Constants {
+        static let cornerRadius: CGFloat = 8.0
+        static let borderStrokeWidth: CGFloat = 2.0
```

Comment:
When I compare the outline buttons in Figma vs reality, the border looks thicker in Figma. However, the inspector shows a 2px border. Not sure why it is 😀

<img width="400" alt="image" src="https://github.com/user-attachments/assets/ce6e25d8-ab5a-4815-b347-607ad458147b" />

<img width="400" alt="image" src="https://github.com/user-attachments/assets/09b7b511-1dc7-4bdd-b332-e56def43d539" />

Reply:
This took me a while and I was tempted to just update the border width value or suspect it's from the screen scale, then I finally found the cause - it was because `stroke` draws half of the border inward and half outward. When this is in the overlay of the button, the outward border gets clipped and thus it looks like half of the border width. In https://github.com/woocommerce/woocommerce-ios/pull/15102/commits/0dacf1429a33d7c30db71dbe450afb0d1aba7dbd, `strokeBorder` is now used to ensure the border is fully drawn inward and visible in the button.

Reply:
Thank you for following through on this! Excellent finding.